### ISSUE TITLE
Rule proposal: detect useless template literal interpolation

### ISSUE THREAD

**Issue by karlhorky** [typescript-eslint/typescript-eslint#2846]
Hi there,

First of all, thanks for this great set of ESLint + TS utilities, these are great!

I am unsure if I'm doing this right, because the issue template was about bug reports, but I would like to propose a new rule. Since I didn't see a template for this, I'm going to go ahead like this, and then can change it / resubmit if necessary.

What I wanted to bring up was the idea of a `@typescript-eslint/no-useless-interpolation` rule, similar to the https://github.com/eslint/eslint/issues/10798 proposal to ESLint, which as far as I can tell, never got implemented.

The idea would be, if there is useless interpolation with a template string, this would be an error.

Example of **incorrect** code:

```jsx
`${'abc'}` // incorrect

const str = 'def';
`${str}` // incorrect

`123${'abc'}` // incorrect

`${'abc'}${'def'}` // incorrect
```

Examples of **correct** code:

```jsx
'abc' // correct

const str = 'def';
str // correct

'123abc' // correct

'abcdef' // correct
```

I would suggest only doing this for strings.

## Remaining template things

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Comment by KurtPreston** [comment]
This rule to be implemented by eslint-plugin-lit
https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-useless-template-literals.md

**Comment by bradzacher** [comment]
@KurtPreston whilst it shares the name - that rule is very different. For starters - it only works on template strings tagged with `html`. Second it doesn't support variables - only literals.

**Comment by rentalhost** [comment]
Would be nice if we can have that!
